scripts/src/packagemapping/generatelocks.py

```diff
@@ -0,0 +1,121 @@
+
+    The filepath is expected to be from the charts directory forward.
+
+    Returns:
+        The compiled regex that groups the category, organization, and chart name.
+
+        E.g. category | organization | chartname
+             partner  | hashicorp    | vault
+    """
+
+    pattern = re.compile(r"charts/(partners|redhat|community)/([\w-]+)/([\w-]+)/OWNERS")
+    return pattern
+
+
+def logError(msg, file=sys.stderr):
+    """logError just prints the msg with an ERROR caption to stderr unless otherwise defined"""
+    print(f"[ERROR] {msg}", file=file)
+
+
+def logInfo(msg, file=sys.stderr):
+    """logError just prints the msg with an INFO caption to stderr unless otherwise defined"""
+    print(f"[INFO] {msg}", file=file)
+
+
+def logWarn(msg, file=sys.stderr):
+    """logWarn just prints the msg with an WARN caption to stderr unless otherwise defined"""
+    print(f"[WARN] {msg}", file=file)
+
+
+def main():
+    """Generates a mapping of chart names to their associated paths.
+
+    Prints the resulting output as a JSON blob.
+
+    Return codes:
+        0:  All is well.
+        10: Parsing failure for the input path to a given OWNERS file.
+        20: One of the expected directory values is empty.
+        30: The OWNERS file at the provided path did not load correctly.
+        40: A duplicate chart name entry has been found.
+        50: The resulting data contained no entries, which
+            is certainly unexpected.
+    """
+    packages = {}
+    for filename in glob("charts/**/OWNERS", recursive=True):
+        logInfo(f"processing file {filename}")
+        pattern = ownerfile_regex()
+        matched = pattern.match(filename)
+        if matched is None:
+            logError(
+                f"did not successfully parse the input. Did you run this in the right place? filename: {filename}",
+            )
+            return 10
+
+        category, organization, chart = matched.groups()
+        if category is None or organization is None or chart is None:
+            logError(
+                f"did not successfully parse the input filename: {filename}",
+                "expecting format charts/{category}/{organization}/{chartname}/OWNERS. Did you run this in the right place?",
+            )
+            return 20
+
+        ownersContentLoaded, ownersContent = owners_file.get_owners_data_from_file(
+            filename
+        )
+        if not ownersContentLoaded:
+            logError(f"Failed to load OWNERS file content. filename: {filename}")
+            return 30
+
+        owners_value_chart_name = owners_file.get_chart(ownersContent)
+        owners_value_vendor_label = owners_file.get_vendor_label(ownersContent)
+
+        if owners_value_chart_name != chart:
+            logError(
+                f"the chart name in the OWNERS file did not match the chart name directory structure. OWNERS_FILE_VALUE={owners_value_chart_name}, DIRECTORY_VALUE:{chart}"
+            )
```

Comment:
On top of logging an Error, do we need to stop here ? Same with the next block, that checks if the vendor label matches the organization name directory structure.

Reply:
We can. I'm trying to think; this will most likely impact Red Hat charts because their chart names will have a different requirement moving forwrad (the prefix). I would need to correct their structure in order to not fail here.

Let's go ahead and throw an error for now. I'll push this fix shortly.

Reply:
This change has been pushed.